### ISSUE TITLE
Pretext file is not loading correctly

### ISSUE THREAD

**Issue by carolhsb** [wtsi-hpag/PretextView#5]
Hi Ed,

I've installed PretextView 0.2.2 for Mac, and after loading the pretext file I got this:

![Screen Shot 2021-07-10 at 23 03 18](https://user-images.githubusercontent.com/71472484/125180701-87f10200-e1d3-11eb-9fc7-cab2b47c4ac7.png)

I know it isn't a problem with the file because I've loaded it on another computer and it worked.
Could you help me?

Thanks,

Carol

**Comment by EdHarry** [comment]
Hi Carol,

Can you confirm the map file is the correct size compared to where you successfully opened it, i.e. it didn’t get truncated or corrupted while transferring it to your computer?

If so, is there a way you could send it to me (google drive, Dropbox etc …)?

**Comment by carolhsb** [comment]
Hi Ed,

I've checked and it isn't corrupted. Give me an email and I will send you a link to the file.

Thanks

**Comment by EdHarry** [comment]
Could you send the link to ***@***.***

**Comment by carolhsb** [comment]
Hi Ed,

I can't see your email. This is the link:

https://drive.google.com/file/d/1XU8chNAHSwPtPLKjlwZYTtOWTIo0S5PW/view?usp=sharing

**Comment by EdHarry** [comment]
Ah, sorry about that, GitHub censors email addresses in comments by default.

The map looks fine to me on my Mac. Could you check your display gamma settings, either from the UI (by pressing the ‘u’ key) or using the left/right arrow keys.

**Comment by carolhsb** [comment]
No problem. 

I've checked the gamma settings but didn't work :(

**Comment by EdHarry** [comment]
Can you try running from command line:

> PretextView.app/Contents/MacOS/PretextView

Do you get any error messages printed to your console?

**Comment by carolhsb** [comment]
I've run it from the command line and didn't get any error messages. The file seems to be loading correctly.
![Screen Shot 2021-07-12 at 12 08 37](https://user-images.githubusercontent.com/71472484/125311732-03ed6600-e30a-11eb-9e9b-23ac6ce86d4d.png)

**Comment by EdHarry** [comment]
Well, to be honest, I'm pretty lost.

Have you been able to open other maps on the same computer successfully?

**Comment by carolhsb** [comment]
I've opened other maps, but from the same data. The result was the same.

Could you share with me another pretext file so I can test opening it?

**Comment by EdHarry** [comment]
Try this:
https://drive.google.com/file/d/1wNOFkh7dWbrneoqq8Ep3dZTEdERzWfrM/view?usp=sharing

**Comment by carolhsb** [comment]
![Screen Shot 2021-07-12 at 12 34 45](https://user-images.githubusercontent.com/71472484/125315587-aa873600-e30d-11eb-8898-9b81efbef8fd.png)

**Comment by carolhsb** [comment]
Maybe a dependency is missing?

**Comment by EdHarry** [comment]
There are no dependencies apart from the system OpenGL libraries.

Could you tell me which version of MacOS are you using?

Could you also try this debug build: https://drive.google.com/file/d/1VDYVEsXCZTXwOUAlRuNCdzRXOsliWVx7/view?usp=sharing

Again, could you run this from the command line and report any logged messages.

**Comment by carolhsb** [comment]
Hi Harry,

My OS version is 10.13.6. I have a MacBook Pro (13 inches, late 2011) and the GPU is Intel HD Graphics 3000 that supports OpenGL 3.3 but doesn't support OpenCL...I think that this might be the problem.

I've tried the debug version, there weren't any error messages but the file was loaded the same way.

**Comment by EdHarry** [comment]
It might be an OpenGL issue on that particular hardware, unfortunately I
don't have easy access to that model macbook to test on. I'll see if I can
virtualise the hardware and discover why it's not working.

**Comment by carolhsb** [comment]
Ok! Many thanks

**Comment by EdHarry** [comment]
Hi @carolhsb, sorry for the delay in getting back to you.

I couldn't virtualise the hardware, but I did manage to get my hands on a late 2011 Macbook Pro to test on. It seems that older Apple graphic drivers have very spotty support for OpenGL features, especially for GLSL shaders, which I use quite extensively in PretextView.

I still haven't been able to get the current version to work, but I did manage to get an older version (0.0.3) to work. This version doesn't feature any map editing support unfortunately.
Hopefully you'll at least be able to open a map:
https://drive.google.com/file/d/1LLqolKclNaTp8yXVPFwISl4hi1l9tdY3/view?usp=sharing
